cli/flags.rs

```diff
@@ -2654,12 +2657,20 @@ fn test_parse(flags: &mut Flags, matches: &clap::ArgMatches) {
   }
 
   let concurrent_jobs = if matches.is_present("jobs") {
+    println!(
+      "{}",
+      crate::colors::yellow("Warning: --jobs flag is deprecated. Use 'DENO_JOBS' environment variable instead."),
+    );
     if let Some(value) = matches.value_of("jobs") {
       value.parse().unwrap()
     } else {
       std::thread::available_parallelism()
         .unwrap_or(NonZeroUsize::new(1).unwrap())
     }
+  } else if let Ok(value) = env::var("DENO_JOBS") {
```

Comment:
I don't think we should do this because not every codebase can handle tests running at the same time. We should probably have a `--parallel` flag like @nayeemrmn suggested and DENO_JOBS would then only be used to inform the amount of parallelism. I will discuss it with some people first though to see if there's support for this.

Reply:
Another reason to have a distinct `--parallel` flag is because we will likely have different result formatting in parallel mode, which suggests it should be API rather than an env config.

Reply:
Alright, so if I understand correctly:
- Deprecate `--jobs` flag
- Introduce `--parallel` flag and default number of used threads to number of available CPUs
- Use `DENO_JOBS` env variable to override number of used threads
- Default number of used threads to 1 if `--jobs` and `--parallel` flags are not passed

Reply:
SGTM

Reply:
@mrkldshv we discussed this in our design meeting and we want to keep the jobs flag, but warn when someone provides a number and suggest they use DENO_JOBS env var with --jobs instead. So what you said, but no parallel flag. Also it should prefer the flag value over the env var.

Reply:
Thanks for the information. Keeping `--jobs` flag sounds good to me. I'll update this PR accordingly.